pkg/minikube/tunnel/registry.go

```diff
@@ -61,7 +61,7 @@ func (r *persistentRegistry) IsAlreadyDefinedAndRunning(tunnel *ID) (*ID, error)
 	}
 
 	for _, t := range tunnels {
-		if t.Route.Equal(tunnel.Route) {
+		if tunnel.MachineName == t.MachineName && t.Route.Equal(tunnel.Route) {
```

Comment:
Please add comment here and also update the PR description and title, right now the PR title suggests this only affects the Test and not minikube itself